Implement exit statuses

<!--
PLEASE FILL IN THE TEMPLATE BELOW, DON'T SIMPLY ERASE EVERYTHING.
ISSUES NOT FOLLOWING THIS TEMPLATE WILL BE CLOSED AUTOMATICALLY

Please read https://github.com/battleman/zoomdl/blob/master/README.md carefully.
-->
**Prerequisite**
<!-- Tick ([ ] -> [x], no space) if you have done the following: -->
- [x] I have updated to the latest version,
- [] (If applicable) I have ensured the password is correct with the flag `-v 0`,
- [x] I have searched the opened and closed issues for a duplicate,

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
I wanted to use zoomdl as part of a bash script I am writing for myself to download videos to a ramdisk and upload it directly to my cloud storage. However, I discovered that even if something fails, zoomdl will only print out error messages to the console but not set an exit status so my script can react properly. Zoomdl will report successful completion to the shell in any case.

**To Reproduce**
<!-- Steps to reproduce the behavior: -->
1. Use a bad/expired zoom recording link
2. Zoomdl will report something like "[ERROR] Video not found in page. Is it login-protected?" in the shell
3. The shell will report that zoomdl terminated successfully with exit code 0 and await your input again
4. So if you want to do some error handling as part of a larger script, you're out of luck

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
Zoomdl is distributed as an executable (not just a raw python script) and thus meant to be run in a shell, so it should also report standard shell exit codes.

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->

**URL (opt)**
<!-- Providing the URL where you encounter the bug greatly helps.
If you wish not to share it publicly, you can send it in PM on reddit
https://reddit.com/u/gsbattleman
 -->
<any expired/invalid zoom recording link>
for example: https://www.zoom.us/rec/play/404

**Logs**
<!-- Please add below the output of the command you launched, with the `-v 0` flag.
Remember to edit out the password if it is confidential! -->

```
~ >  ./zoomdl -u https://www.zoom.us/rec/play/404 -v 0
Using standard Windows UA
Changing page to https://www.zoom.us/rec/play/404
Advanced meta failed
Unable to extract chatList from page
Unable to extract transcriptList from page
Metas are {'cookie_currency': 'EUR', 'currency_symbol': '€', '__platformCheck': '', 'detect': 'https://st2.zoom.us/cdn-detect.png', 'chatList': [], 'transcriptList': []}
No video URL in meta, going bruteforce
[ERROR] Video not found in page. Is it login-protected? 
Try to refresh the webpage, and export cookies again
Unable to find metadata, aborting.

~ > echo $?
0
```
**Versions (please complete the following information):**
 - OS: macOS Big Sur 11.6
 - Python Version: homebrew python@3.9 (3.9.13)

**Additional context (opt)**
I want this line to work as intended in my shell script:
` ${script_dir}/zoomdl -u "$url" -f "$filename" || continue`
If zoomdl encounters an error, it should return a non-zero exit code to the shell such that the `continue` statement is executed and skips the rest of the loop this line is contained in.

--- a/zoom_dl/__init__.py
+++ b/zoom_dl/__init__.py
@@ -17,3 +17,5 @@ def main():
     all_urls = [args.url]  # prepare for batch reading
     zdl = ZoomDL(args)
     zdl.download(all_urls)
+
+    sys.exit(zdl.exit_code)

--- a/zoom_dl/zoomdl.py
+++ b/zoom_dl/zoomdl.py
@@ -26,6 +26,7 @@ class ZoomDL():
         self.url, self.domain, self.subdomain = "", "", ""
         self.metadata = None
         self.session = requests.session()
+        self.exit_code = 0
 
         self.loglevel = self.args.log_level
 
@@ -68,6 +69,11 @@ class ZoomDL():
         """
         if level < 5 and level >= self.loglevel:
             print(message)
+
+        """Set system exit code to 1 if zoomdl encounters a (critical) error."""
+        # there is potential here for more granular exit codes if desired
+        if level < 5 and level >= 3:
+            self.exit_code = 1
 
     def _change_page(self, url):
         """Change page, with side methods."""

